woocommerce/woocommerce-gutenberg-products-block
# ⚛️  Introduce Product Element Blocks

This PR moves Product Elements Blocks to the top level so they can be used standalone.

Closes #2702

Product Element Blocks can now be found in the main block inserter in their own section:

![Edit Page ‹ one wordpress test — WordPress 2020-07-16 18-08-57](https://user-images.githubusercontent.com/90977/87701222-84c56080-c78f-11ea-9690-74095c67719c.png)

When inserted outside of the content of a parent block only, you can select a product:
![Edit Page ‹ one wordpress test — WordPress 2020-07-16 18-10-24](https://user-images.githubusercontent.com/90977/87701287-a292c580-c78f-11ea-8c55-85c7a73c26d9.png)

On the frontend, a new `renderStandalone` function converts block markup to components. Looking for feedback here @senadir @nerrad because right now I placed it in single-product block so it runs last—can you suggest a better way to handle this?

**How to test**

Standalone testing:

1. Build, then insert various element blocks into a post. Select products. Ensure content is rendered.
2. Save the post. View the frontend. Confirm again that content is rendered.

Single Product block:

1. Insert single product block and confirm elements still work.
2. Insert more elements into the layout. You should not see a product selector.
3. Save and test rendering.

All products block:

Repeat the test above, ensuring you also test editing the layout, to ensure there are no regressions.

**Follow ups**

- When there is no content (e.g. tag list when there are no tags) we will need to add a placeholder to the editor so the user can see where the block is placed.
- Need feedback on standalone rendering (comment above).
